ai_diffusion/ui/widget.py

```diff
@@ -243,6 +256,8 @@ def _check_is_installed(self):
         self.add_pose_button.setEnabled(self._is_vector_layer())
         self.strength_spin.setVisible(is_installed)
         self.strength_spin.setEnabled(self._is_first_image_mode())
+        self.end_spin.setVisible(is_installed and settings.show_control_end)
+        self.end_spin.setEnabled(self._is_first_image_mode())
```

Comment:
This isn't needed, or rather I think we shouldn't show the end spin box at all for "image" control layer.
That is, add `and mode is not ControlMode.image` to the `setVisible` condition.

Note: ControlMode.image is how IP-adapter is exposed in the UI, it's not technically a ControlNet and applied differently.

Reply:
> Note: ControlMode.image is how IP-adapter is exposed in the UI, it's not technically a ControlNet and applied differently.

Oh, I see. I was not sure what was behind `ControlMode.image`. I'll do the modification.

On a side note, a name change for this particular control could clarify it's usage for anyone wondering. 😅